java/org/apache/jasper/compiler/Parser.java

```diff
@@ -1471,8 +1471,8 @@ private void parseBody(Node parent, String tag, String bodyType) throws JasperEx
                 err.jspError(start, "jasper.error.emptybodycontent.nonempty", tag);
             }
         } else if (bodyType.equalsIgnoreCase(TagInfo.BODY_CONTENT_JSP) ||
-                bodyType.equalsIgnoreCase(TagInfo.BODY_CONTENT_SCRIPTLESS) || (bodyType == JAVAX_BODY_CONTENT_PARAM) ||
-                (bodyType == JAVAX_BODY_CONTENT_TEMPLATE_TEXT)) {
+                bodyType.equalsIgnoreCase(TagInfo.BODY_CONTENT_SCRIPTLESS) || (bodyType.equals(JAVAX_BODY_CONTENT_PARAM)) ||
+                (bodyType.equals(JAVAX_BODY_CONTENT_TEMPLATE_TEXT))) {
```

Comment:
The only time this will be true is when this particular constant is being used, internally. Using .equals is needless pedantry.